tests/test_routes.py

```diff
@@ -0,0 +1,126 @@
+from app.models.planet import Planet
+
+def test_get_all_planets_with_no_records(client):
+    # Act
+    response = client.get("/planets")
+    response_body = response.get_json()
+
+    # Assert
+    assert response.status_code == 200
+    assert response_body == []
+
+def test_get_all_planets_with_fixture_records(client, many_planets):
+    # Act
+    response = client.get("/planets")
+    response_body = response.get_json()
+
+    # Assert
+    assert response.status_code == 200
+    assert len(response_body) == 4
+    #assert
+
+
+def test_get_one_planet_no_fixture(client):
+    # Act
+    response = client.get("/books/1")
+    response_body = response.get_json()
+
+    # Assert
+    assert response.status_code == 404
+    assert response_body is None
+
+def test_get_one_planet(client, one_planet):
+    # Act
+    response = client.get(f"/planets/{one_planet.id}")
+    response_body = response.get_json()
+
+    # Assert
+    assert response.status_code == 200
+    assert response_body == {
+        "id": 1,
+        "name": "Planet number one",
+        "description": "watr 4evr",
+        "color": "color"
+    }
+
+def test_get_one_planet_from_many_planets_valid_id(client, many_planets):
+    #Arrange
+    EXPECTED_ID = 3
+    
+    # Act
+    response = client.get(f"/planets/{EXPECTED_ID}")
+    response_body = response.get_json()
+
+    # Assert
+    assert response.status_code == 200
+    assert response_body == {
+        "id": 3,
+        "name": "Planet of butterflies",
+        "description": "Butterflies and unicorns live there",
+        "color": "Rainbow"
+    }
+
+def test_get_one_planet_from_many_planets_non_existing_id_returns_404(client, many_planets):
+    #Arrange
+    NON_EXISTING_ID = 7
+    
+    # Act
+    response = client.get(f"/planets/{NON_EXISTING_ID}")
+    response_body = response.get_data(as_text=True)
+
+    # Assert
+    assert response.status_code == 404
+    assert response_body == "{\"message\":\"planet 7 not found\"}\n"
+
+def test_get_one_planet_from_many_planets_invalid_id_returns_400(client, many_planets):
+    #Arrange
+    INVALID_ID = "ABC"
+    
+    # Act
+    response = client.get(f"/planets/{INVALID_ID}")
+    response_body = response.get_data(as_text=True)
+
+    # Assert
+    assert response.status_code == 400
+    assert response_body == "{\"message\":\"planet ABC has an invalid planet_id\"}\n"
+
+def test_create_planet_can_create_planet_in_empty_db(client):
```

Comment:
Have a create request with an invalid request body would be a good test as well.